Tests/SonarScanner.MSBuild.PreProcessor.Test/JreCaching/JreCacheTests.cs

```diff
@@ -469,17 +467,21 @@ public async Task Checksum_DownloadFilesChecksumDoesNotMatchExpectation(string f
         directoryWrapper.Exists(sha).Returns(true);
         fileWrapper.Exists(file).Returns(false);
         fileWrapper.Create(Arg.Any<string>()).Returns(new MemoryStream());
+        fileWrapper.When(x => x.Delete(file)).Do(x => throw new FileNotFoundException());
         var fileStream = new MemoryStream();
         fileWrapper.Open(file).Returns(fileStream);
         checksum.ComputeHash(fileStream).Returns(fileHashValue);
-
         var sut = CreateSutWithSubstitutes();
         var result = await sut.DownloadJreAsync(home, new("filename.tar.gz", expectedHashValue, "javaPath"), () => Task.FromResult<Stream>(new MemoryStream()));
+
         result.Should().BeOfType<JreCacheFailure>().Which.Message.Should().Be("The checksum of the downloaded Java runtime environment does not match the expected checksum.");
         testLogger.AssertDebugLogged($"The checksum of the downloaded file is '{fileHashValue}' and the expected checksum is '{expectedHashValue}'.");
+        testLogger.AssertDebugLogged("Deleting mismatched JRE Archive.");
+        testLogger.AssertDebugLogged("Failed to delete mismatched JRE Archive. Unable to find the specified file.");
```

Comment:
I think it is better to assert the whole debug log and therefore I changed from this assertion style to something like this for most of my later tests:
```suggestion
        testLogger.DebugMessages.Should().BeEquivalentTo(
            $"The checksum of the downloaded file is '{fileHashValue}' and the expected checksum is '{expectedHashValue}'.",
            "Deleting mismatched JRE Archive.",
            "Failed to delete mismatched JRE Archive. Unable to find the specified file.");
```
You can also use some of the other collection assertions like `ContainInOrder`